salt/_modules/kubernetes.py

```diff
@@ -1797,3 +1797,132 @@ def replace_role(
             raise CommandExecutionError(exc)
     finally:
         _cleanup(**cfg)
+
+
+def node_annotations(name, **kwargs):
+    '''
+    Return the annotations of the node identified by the specified name
+
+    CLI Examples::
+
+        salt '*' kubernetes.node_annotations name="minikube"
+    '''
+    match = node(name, **kwargs)
+
+    if match is not None:
+        return match['metadata']['annotations']
+
+    return {}
+
+
+def node_add_annotation(node_name, annotation_name, annotation_value, **kwargs):
+    '''
+    Set the value of the annotation identified by `annotation_name` to
+    `annotation_value` on the node identified by the name `node_name`.
+    Creates the annotation if not present.
+
+    CLI Examples::
+
+        salt '*' kubernetes.node_add_annotation node_name="minikube" \
+            annotation_name="foo" annotation_value="bar"
+    '''
+    cfg = _setup_conn(**kwargs)
+    try:
+        api_instance = kubernetes.client.CoreV1Api()
+        body = {
+            'metadata': {
+                'annotations': {
+                    annotation_name: annotation_value
+                }
+            }
+        }
+        api_response = api_instance.patch_node(node_name, body)
+        return api_response
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception('Exception when calling CoreV1Api->patch_node')
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
+
+    return None
+
+
+def node_remove_annotation(node_name, annotation_name, **kwargs):
+    '''
+    Removes the annotation identified by `annotation_name` from
+    the node identified by the name `node_name`.
+
+    CLI Examples::
+
+        salt '*' kubernetes.node_remove_annotation node_name="minikube" \
+            annotation_name="foo"
+    '''
+    cfg = _setup_conn(**kwargs)
+    try:
+        api_instance = kubernetes.client.CoreV1Api()
+        body = {
+            'metadata': {
+                'annotations': {
+                    annotation_name: None}
+                }
+        }
+        api_response = api_instance.patch_node(node_name, body)
+        return api_response
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception('Exception when calling CoreV1Api->patch_node')
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
+
+    return None
+
+
+def node_taints(node_name, **kwargs):
+    match = node(node_name, **kwargs)
+
+    if match is not None:
+        return match['spec']['taints']
+
+    return None
```

Comment:
This means the node doesn't exist, rather than 'no taints'. If this is how other procedures in the module work, then fine. We should rework it anyway :stuck_out_tongue:

Reply:
Actually I should have returned an empty list instead (copy-pasted the None as for the other methods I guess), but yes this module needs a rework :)